### PR TITLE
[candidate_parameters] Update test plan and README to include widgets

### LINKED ISSUE #6433: [candidate_parameters] Test plan and README do not describe widgets
The test plan for the candidate_parameters module registers widgets into the candidate_profile, but the README does not describe the interaction and the test plan does not test the widget.

### PR DESCRIPTION
**Pull request by driusan**
Update the test plan to include the widgets registered in other modules.

Fixes #6433

### CODE REVIEW

File: modules/candidate_parameters/README.md [hunk 1]
```diff
@@ -64,3 +64,5 @@ using the `CouchDB_Import_Demographics.php` tool script. Any modifications or ad
 to this module should be manually added to the script in order to be included 
 in the DQT Demographics table. Alternatively, these modifications or additions can be 
 added to a separate script that is then exported to the DQT.
+- The `candidate_parameters` module registers widgets in the `candidate_profile`
```
Comment (christinerogers) [inline modules/candidate_parameters/README.md:67]:
```suggestion
- The `candidate_parameters` module appears as a widget in the `candidate_profile`
```
in terms of describing how this is seen by both users and admins, describing how it pops up is probably more understandable than the architectural description that it "registers modules"

File: modules/candidate_parameters/test/TestPlan.md [hunk 2]
```diff
@@ -81,3 +81,37 @@
 38. Check that entering a date of death before the date of birth gives a data entry error.
 39. Check that entering a date later than today gives a data entry error.
 40. Check that the date stored matches the _DOD Format_ specified in the configuration module.
+
+## Widgets
+41. Access a candidate in the `candidate_profile` module
+
+### Candidate Info Widget terms
+42. Within the "Candidate Info" card, after the information which
```
Comment (christinerogers) [inline modules/candidate_parameters/test/TestPlan.md:89]: point 42 : How is the user to test this? 
If it's not testable, it's just pre-amble for what follows, and shouldn't get a number.

File: modules/candidate_parameters/test/TestPlan.md [hunk 3]
```diff
@@ -81,3 +81,37 @@
...
+    `candidate_parameters` module, and that the term does *not*
+    appear if the candidate does not have a caveat.
+45. Verify that any terms from the "Candidate Parameters"
+    `parameter_type_category` in step 5 of the test plan are displayed
+    in the "Candidate Info" card.
+
+### Consent Widget
+46. With Consent configured in LORIS, verify that there is a
+   "Consent" card in the Candidate Profile widget.
+47. Verify that the card does *not* appear is Consent is
```
Comment (christinerogers) [inline modules/candidate_parameters/test/TestPlan.md:106]:
```suggestion
47. Verify that the card does *not* appear if Consent is
```

File: modules/candidate_parameters/test/TestPlan.md [hunk 4]
```diff
@@ -81,3 +81,37 @@
...
+    description of the reason for a caveat which matches the
+    Caveat Emptor flag of the "Candidate Info" tab of the
+    `candidate_parameters` module, and that the term does *not*
+    appear if the candidate does not have a caveat.
+45. Verify that any terms from the "Candidate Parameters"
+    `parameter_type_category` in step 5 of the test plan are displayed
+    in the "Candidate Info" card.
+
+### Consent Widget
+46. With Consent configured in LORIS, verify that there is a
```
Comment (christinerogers) [inline modules/candidate_parameters/test/TestPlan.md:104]: there's something unclear about "configured", and configured/enabled should be consistent across points 46-49 so i'm suggesting `enabled`. 
```suggestion
46. With Consent enabled in this LORIS, verify that there is a
```

File: modules/candidate_parameters/test/TestPlan.md [hunk 5]
```diff
@@ -81,3 +81,37 @@
...
+   "Consent" card in the Candidate Profile widget.
+47. Verify that the card does *not* appear is Consent is
+    not configured in LORIS.
```
Comment (christinerogers) [inline modules/candidate_parameters/test/TestPlan.md:107]:
```suggestion
    not enabled in LORIS.
```

File: modules/candidate_parameters/test/TestPlan.md [hunk 6]
```diff
@@ -81,3 +81,37 @@
...
+48. With Consent enabled, ensure that all types of Consent
+    are displayed in a table in the card, regardless of
+    whether or not the candidate being viewed has any
+    data for that consent type.
+49. Verify that rows for both "Yes" and "No" are displayed
```
Comment (christinerogers) [inline modules/candidate_parameters/test/TestPlan.md:112]: fyi - "no" will be displayed first if/when this is reactified (given sort order is unspecified)

File: modules/candidate_parameters/test/TestPlan.md [hunk 7]
```diff
@@ -81,3 +81,37 @@
...
+    properly in the table and match the "Consent" tab of
+    the module.
+50. Verify that a consent type which has no data saved
+    for the candidate is displayed with an empty row in
+    the table containing only a dash.
```
Comment (christinerogers) [inline modules/candidate_parameters/test/TestPlan.md:117]:
```suggestion
    the table; the row should display only a dash.
```
punctuating to un-suggest the table (vs row) contains a dash

File: modules/candidate_parameters/test/TestPlan.md (hunk 4 above)
Comment (driusan) [reply modules/candidate_parameters/test/TestPlan.md:104]: As far as I know there is no "enable/disable consent" option in LORIS. It's either configured (there are consent types set up) or it's not.

File: modules/candidate_parameters/test/TestPlan.md (hunk 6 above)
Comment (driusan) [reply modules/candidate_parameters/test/TestPlan.md:112]: There's no sorting. It's a plain table with a column for Yes/No.

File: modules/candidate_parameters/test/TestPlan.md (hunk 5 above)
Comment (driusan) [reply modules/candidate_parameters/test/TestPlan.md:107]: See other comment.. there is no enable/disable consent in LORIS as far as I know.

Comment (christinerogers) [reply modules/candidate_parameters/test/TestPlan.md:110]: Ok.  I don't think it's clearer now but at least it's consistent 👍

**Pull request merged**